File: src/Java-Kotlin/SampleJavaBasic/src/org/csystem/math/geometry/PointF.java
/*----------------------------------------------------------------------------------------------------------------------
	PointF sınıfı
----------------------------------------------------------------------------------------------------------------------*/
package org.csystem.math.geometry;

import static java.lang.Math.sqrt;

public class PointF {
	public float x;
	public float y;
	
	public PointF()
	{		
	}
	
	public PointF(float x)
	{
		this.x = x;
	}
	
	public PointF(float x, float y)
	{
		this.x = x;
		this.y = y;
	}
	
	public double distance(Point p)
	{
		return this.distance(p.x, p.y);
	}
	
	public double distance(float a, float b)
	{
		return sqrt((x - a) * (x - a) + (y - b) * (y - b));								
	}
	
	public void offset(float dxy)
	{
		this.offset(dxy, dxy);
	}
	
	public void offset(float dx, float dy)
	{
		x += dx;
		y += dy;
	}
	
	public String toString()
	{
		return String.format("{x: %f, y: %f}", x, y);
	}	
}

File: src/Java-Kotlin/Sample/src/main/java/org/csystem/util/math/geometry/PointF.kt
/*----------------------------------------------------------------------------------------------------------------------
    PointF sınıfı
----------------------------------------------------------------------------------------------------------------------*/
package org.csystem.util.math.geometry

import kotlin.math.sqrt

data class PointF(var x: Float = 0F, var y: Float = 0F) {
    constructor(x: Int, y: Int) : this(x.toFloat(), y.toFloat())
    constructor(x: Long, y: Long) : this(x.toFloat(), y.toFloat())

    fun distance(p: PointF) = distance(p.x, p.y)

    fun distance(x: Int, y: Int) = distance(x.toFloat(), y.toFloat())
    fun distance(x: Float = 0F, y: Float = 0F)
            = sqrt((this.x - x.toDouble()) * (this.x - x.toDouble()) + (this.y - y.toDouble()) * (this.y - y.toDouble()))

    fun offset(dx: Float, dy: Float = dx)
    {
        x += dx
        y += dy
    }
}